web/src/app/actions/hooks.ts

```diff
@@ -1,46 +1,111 @@
-import { urlParamSelector, urlPathSelector } from '../selectors'
-import { setURLParam as setParam, resetURLParams as resetParams } from './main'
-import { useSelector, useDispatch } from 'react-redux'
-import { warn } from '../util/debug'
-import joinURL from '../util/joinURL'
-import { pathPrefix } from '../env'
+import { History, Location } from 'history'
+import { useHistory, useLocation } from 'react-router-dom'
 
 export type Value = string | boolean | number | string[]
 
+// sanitizeURLParam serializes a value to be ready to store in a URL.
+// If a value cannot or should not be stored, an empty string is returned.
+export function sanitizeURLParam(value: Value): string | string[] {
+  if (Array.isArray(value)) {
+    const filtered = value.map((v) => v.trim()).filter(Boolean)
+    if (filtered.length === 0) return ''
+    return filtered
+  }
+
+  switch (typeof value) {
+    case 'string':
+      return value.trim()
+    case 'boolean':
+      if (value === true) return '1'
+      return ''
+    case 'number':
+      return value.toString()
+    default:
+      return ''
+  }
+}
+
+// getURLParam gets the URL param value and converts it from string|string[]
+// to the desired type based on the provided default value.
+export function getURLParam<T extends Value>(
+  url: URLSearchParams,
+  name: string,
+  defaultValue: T,
+): T {
+  if (!url.has(name)) return defaultValue
+  if (Array.isArray(defaultValue)) return url.getAll(name) as T
+  if (typeof defaultValue === 'boolean') return (url.get(name) === '1') as T
+  if (typeof defaultValue === 'string') return (url.get(name) || '') as T
+  if (typeof defaultValue === 'number') return +(url.get(name) as string) as T
+  return defaultValue
+}
+
+// setURL will replace the latest browser history entry with the provided config.
+function setURL(
```

Comment:
nit: can we call this `setParams` or similar?

Also, order the args by importance/change frequency: history, location, params
history is our fixed API for updating, we use location as a reference, and params is the real argument we're using to update the search string.